tests/unit/test_middleware/test_csrf_middleware.py

```diff
@@ -241,6 +249,18 @@ def post_handler2(data: dict = Body(media_type=RequestEncodingType.URL_ENCODED))
         assert response.status_code == HTTP_201_CREATED
         assert response.json() == data
 
+    with create_test_client(
```

Comment:
This should be a separate test, with a descriptive name. You can also add a link to the GH issue as a comment there. That makes it easier to track why things are being tested